docs/architecture/adr-021-subspace-specific-custom-fee-tokens.md

```diff
@@ -0,0 +1,152 @@
+# ADR 021: Subspace specific custom fee tokens
+
+## Changelog
+
+- April 24th, 2023: First draft;
+
+## Status
+
+Proposed
+
+## Abstract
+
+This ADR introduces a mechanism that allows users to pay fees within a designated subspace using their existing service tokens.
+
+## Context
+
+Desmos uses subspaces to represent virtual applications built on top of our protocol. Currently, subspaces are mainly used to store content such as posts, reports, and reactions. While this is sufficient for developing DApps on Desmos, it may not be enough to convince existing applications to migrate to our platform since Desmos is lack of a mechanism that enables users to pay fees with existing service tokens.
+
+For instance, consider Keplr's on-chain governance discussions feature, which is currently implemented using a custom backend that stores data in a centralized way. Although Keplr could easily use Desmos as their backend, they are not willing to do so because it offers no advantage to them. Migrating to Desmos would simply transfer their existing users to our chain without any benefits for them.
+
+## Decision
+
+To address the issue mentioned above, we propose implementing a mechanism that allows users to pay fees using the existing service tokens within the specified subspace. The mechanism must have the following features:
+- Defining a minimum gas prices for non-DSM tokens that validators are willing to accept via governance proposal;
+- Allowing the subspace manager(s) to update the list of fee tokens allowed in the subspace;
+- Checking the on-chain minimum gas price requirement for non-DSM fees using a custom `TxChecker`.
+
+The upcoming changes are as follows:
+
+### Types
+
+We will define a new `Params` type to show the `x/subspaces` parameters.
+
+```proto
+// Params contains the module parameters
+message Params {
+  // List of the minimum gas prices to be accepted by validators
+  repeated Coin min_gas_prices = 2;
+}
+```
```

Comment:
Why is this needed? Each subspace will have its own `allowed_fee_tokens` list anyway. There's no need to have a module-wide list